pkg/datapath/linux/sysctl/ops.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package sysctl
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/reconciler"
+)
+
+func newOps(log logrus.FieldLogger, fs afero.Fs, cfg Config) reconciler.Operations[*tables.Sysctl] {
+	return &ops{log: log, fs: fs, procFs: cfg.ProcFs}
+}
+
+type ops struct {
+	log    logrus.FieldLogger
+	fs     afero.Fs
+	procFs string
+}
+
+func (ops *ops) Update(ctx context.Context, txn statedb.ReadTxn, s *tables.Sysctl, changed *bool) error {
+	log := ops.log.WithFields(logrus.Fields{
+		logfields.SysParamName:  s.Name,
+		logfields.SysParamValue: s.Val,
+	})
+
```

Comment:
Shouldn't we set `changed` to false here? Otherwise this is incorrect:
```
.../sysctl-path = X
Update(.../sysctl-path, X, &changed) // say, changed was true before call
sysctl is not changed, but changed=true for the caller
```

Reply:
I believe the current code does exactly what the interface describes. If `changed` is non-nil, its false by default and we only need to change it to true if we did make a change. And the function first does a read before we write and quiets before changed=true if there is nothing to do.